src/webui/api/torrentscontroller.cpp

```diff
@@ -856,3 +856,18 @@ void TorrentsController::removeCategoriesAction()
     for (const QString &category : categories)
         BitTorrent::Session::instance()->removeCategory(category);
 }
+
+void TorrentsController::categoriesAction()
+{
+    QVariantHash categories;
```

Comment:
You don't need such intermediate containers in this action. Just use QJsonObject directly.

Reply:
Switched to directly using `QJsonObject`